hirte/src/common/parse-util.c

```diff
@@ -40,3 +42,128 @@ bool parse_port(const char *in, uint16_t *ret) {
         *ret = (uint16_t) l;
         return true;
 }
+
+bool is_IP(char *str) {
+        long num = 0;
+        int dots = 0;
+        char *ptr = NULL;
+
+        if (str == NULL) {
+                return false;
+        }
+
+        ptr = str;
+        // NOLINTNEXTLINE
+        while (*ptr != '\0') {
+                if (*ptr == '.') {
+                        dots++;
+                } else if (!isdigit(*ptr)) {
+                        return false;
+                }
+                ptr++;
+        }
+
+        if (dots != 3) {
+                return false;
+        }
+        int const max = 255;
+        ptr = strtok(str, ".");
+        int const base = 10;
+        while (ptr != NULL) {
+                num = strtol(ptr, NULL, base);
+                if (num >= 0 && num <= max) {
+                        ptr = strtok(NULL, ".");
+                } else {
+                        return false;
+                }
+        }
+
+        return true;
+}
+
+bool is_port(char *str) {
+        long port = 0;
+        int const max_port = 65535;
+
+        if (str == NULL) {
+                return 0;
+        }
+        int const base = 10;
+        port = strtol(str, NULL, base);
+        if (port < 0 || port > max_port) {
+                return 0;
+        }
+        return 1;
+}
+
+bool is_IPs(char *str) {
```

Comment:
Same as for `is_IP`. Instead of only testing if each element in the list (e.g. `127.0.0.1, 128.0.0.1`) is a valid IP, I'd directly parse them. For example:
```C
uint16_t *parse_ips_from_string(const char *ip_string, const char delimiter){
   // count number of delimiter occurences
   uint16_t ips[number of occ];
   // parsing each ip in ip_string
}
```